qiskit/pulse/commands/parametric_pulses.py

```diff
@@ -0,0 +1,261 @@
+from qiskit.pulse.pulse_lib.discrete import gaussian, gaussian_square, drag, constant
+
+from .command import Command
+from .sample_pulse import SamplePulse
+from .instruction import Instruction
+
+# pylint: disable=missing-docstring
+
+
+class ParametricPulse(Command):
+    """The abstract superclass for parametric pulses."""
+    prefix = 'p'
+
+    @abstractmethod
+    def __init__(self, duration: int):
+        """Create a parametric pulse command.
+
+        Args:
+            duration: Pulse length in terms of the the sampling period `dt`.
+        """
+        super().__init__(duration=duration)
+
+    @abstractmethod
+    def get_sample_pulse(self) -> SamplePulse:
+        """Return a SamplePulse with samples filled according to the formula that the pulse
+        represents and the parameter values it contains.
+        """
+        pass
+
+    def get_params(self) -> Dict[str, Any]:
+        """Return a dictionary containing the parameters specific to this parametric pulse."""
+        return {attr[1:]: getattr(self, attr)
+                for attr in self.__dict__
+                if attr.startswith('_') and attr != '_name'}
+
+    def to_instruction(self, channel: PulseChannel,
+                       name: Optional[str] = None) -> 'ParametricInstruction':
+        # pylint: disable=arguments-differ
+        return ParametricInstruction(self, channel, name=name)
+
+    def __repr__(self):
+        return '{}(name={}, params={})'.format(self.__class__.__name__,
+                                               self.name,
+                                               self.get_params())
+
+
+class Gaussian(ParametricPulse):
+    """
+    A truncated pulse envelope shaped according to the Gaussian function whose mean is centered at
+    the center of the pulse (duration / 2):
+
+        f(x) = amp * exp( -(1/2) * (x - duration/2)^2 / sigma^2) )  ,  0 <= x < duration
+    """
+
+    def __init__(self,
+                 duration: int,
+                 amp: complex,
+                 sigma: float):
+        """Initialize the gaussian command.
+
+        Args:
+            duration: Pulse length in terms of the the sampling period `dt`.
+            amp: The amplitude of the Gaussian envelope.
+            sigma: A measure of how wide or narrow the Gaussian peak is; described mathematically
+                   in the class docstring.
+        """
+        self._amp = complex(amp)
+        self._sigma = sigma
+        super().__init__(duration=duration)
+
+    @property
+    def amp(self):
+        return self._amp
+
+    @property
+    def sigma(self):
+        return self._sigma
+
+    def get_sample_pulse(self) -> SamplePulse:
+        return gaussian(duration=self.duration, amp=self.amp,
+                        sigma=self.sigma, zero_ends=False)
+
+
+class GaussianSquare(ParametricPulse):
+    """
+    A square pulse with a Gaussian shaped risefall on either side:
+
+        risefall = duration - width / 2
+
```

Comment:
I width is the length of the square part then should this not be `risefall = (duration - width)/2`?

Reply:
absolutely, will update